### PR TITLE
Implement `VirtualMachine.DeduceOp0`

### PR DESCRIPTION
**Pull request by fmoletta**

### CODE REVIEW

File: pkg/vm/memory/relocatable.go [hunk 1]
```diff
@@ -152,7 +176,7 @@ func (m MaybeRelocatable) AddMaybeRelocatable(other MaybeRelocatable) (MaybeRelo
 		other_felt, _ := other.GetFelt()
 		relocatable, err := m_rel.AddFelt(other_felt)
 		if err != nil {
-			return MaybeRelocatable{}, nil
+			return *NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(0)), err
```
Comment (gabrielbosio) [inline pkg/vm/memory/relocatable.go]: We can use `FeltZero` instead of `FeltFromUint64(0)`, Right?

File: pkg/vm/memory/relocatable.go [hunk 2]
```diff
@@ -161,10 +185,48 @@ func (m MaybeRelocatable) AddMaybeRelocatable(other MaybeRelocatable) (MaybeRelo
 		m_felt, _ := m.GetFelt()
 		relocatable, err := other_rel.AddFelt(m_felt)
 		if err != nil {
-			return MaybeRelocatable{}, err
+			return *NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(0)), err
```
Comment (gabrielbosio) [inline pkg/vm/memory/relocatable.go]: Same here

File: pkg/vm/memory/relocatable.go [hunk 3]
```diff
@@ -161,10 +185,48 @@ func (m MaybeRelocatable) AddMaybeRelocatable(other MaybeRelocatable) (MaybeRelo
...
 		}
 		return *NewMaybeRelocatableRelocatable(relocatable), nil
 	} else {
-		return MaybeRelocatable{}, errors.New("RelocatableAdd")
+		return *NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(0)), errors.New("RelocatableAdd")
```
Comment (gabrielbosio) [inline pkg/vm/memory/relocatable.go]: Same here

File: pkg/vm/memory/relocatable.go [hunk 4]
```diff
@@ -161,10 +185,48 @@ func (m MaybeRelocatable) AddMaybeRelocatable(other MaybeRelocatable) (MaybeRelo
...
+		return *NewMaybeRelocatableRelocatable(relocatable), nil
+
+	} else if is_rel_m && is_rel_other {
+		offset_diff, err := m_rel.Sub(other_rel)
+		if err != nil {
+			return *NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(0)), err
+		}
+		return *NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(uint64(offset_diff))), nil
+	} else {
+		return *NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(0)), errors.New("Cant sub Relocatable from Felt")
```
Comment (gabrielbosio) [inline pkg/vm/memory/relocatable.go]: Same here

**Review by gabrielbosio — APPROVED** [review]
Looks good, I left some minor comments

**Review by pablodeymo — APPROVED** [review]
🚀

**Pull request merged**